moveit_simple/src/moveit_simple.cpp

```diff
@@ -799,14 +803,34 @@ trajectory_msgs::JointTrajectoryPoint Robot::toJointTrajPtMsg(
 
 
 
+std::vector<double> Robot::getJointState(void) const
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+  ros::spinOnce();
+  std::vector<double> current_joint_positions;
+  virtual_robot_state_->update();
+  virtual_robot_state_->copyJointGroupPositions(joint_group_->getName(), current_joint_positions);
+  return current_joint_positions;
+}
+
+
 
-void Robot::updateState(const sensor_msgs::JointStateConstPtr& msg)
+std::vector<double> OnlineRobot::getJointState(void) const
```

Comment:
I think making the getJointState a protected method in Robot and public in OnlineRobot makes more sense. Because JointState in Robot doesn't make much sense for user.